src/steps/ms-defender/client.ts

```diff
@@ -57,4 +64,48 @@ export class DefenderClient extends GraphClient {
       link: `${this.BASE_URL_API}/machines/${machineId}?`,
     });
   }
+
+  private async iterateSecurityResource<T>({
+    resourceUrl,
+    callback,
+    limit = 50,
+  }: {
+    resourceUrl: string;
+    callback: (item: T) => void | Promise<void>;
+    limit?: number;
+  }): Promise<void> {
+    let retries = 0;
+    let response: GraphClientResponse<T> | undefined;
+    let skip = 0;
+    let objectsSeen = 0;
+    do {
+      try {
+        response = await this.callApiWithRetry<T>({
+          link: resourceUrl,
+          query: { $top: limit, $skip: skip },
+        });
+      } catch (err) {
+        if (
+          err.message ===
+            'CompactToken parsing failed with error code: 80049217' &&
+          retries < 5
+        ) {
+          // Retry a few times to handle sporadic timing issue with this sdk - https://github.com/OneDrive/onedrive-api-docs/issues/785
+          retries++;
+          continue;
+        } else {
+          this.handleApiError(err, resourceUrl);
+          break;
+        }
+      }
+
+      if (response) {
+        for (const value of response.value) {
+          await callback(value);
+        }
+        objectsSeen += response.value.length;
+      }
+      skip += limit;
+    } while (skip == objectsSeen);
```

Comment:
If there were exactly 200 objects to iterate over, could this allow for an extra request _that'd fail_? Or would it just respond with an empty collection?